blscurve/miracl/miracl_min_pubkey_sig_core.nim

```diff
@@ -60,6 +60,10 @@ type
     ## Long-term storage of this key also requires adequate protection.
     ##
     ## At the moment, the nim-blscurve library does not guarantee such protections
+    ##
+    ## Guarantees:
+    ## - SecretKeys are always created (via hkdf_mod_r) or deserialized (via `fromBytes`)
```

Comment:
we can probably slap a `requiresInit` on it to catch any zero-initialized stragglers